.github/workflows/build-container.yml

```
@@ -33,20 +34,23 @@ jobs:
       env:
         DVER: ${{ matrix.dver }}
         REPO: ${{ matrix.repo }}
+        SERIES: ${{ matrix series }}
         TIMESTAMP: ${{ needs.make-date-tag.outputs.dtag }}
       run: |
-        TAG_PREFIXES=("el$DVER-")
-        TAG_NAMES="$REPO $REPO-$TIMESTAMP"
-        [[ $DVER == 7 ]] && TAG_PREFIXES+=('')
-        [[ $REPO == 'testing' ]] && TAG_NAMES="$TAG_NAMES fresh $TIMESTAMP"
+        IMAGE_REPO="${GITHUB_REPOSITORY/opensciencegrid\/docker-/opensciencegrid/}"
+        TAG_PREFIXES=("{$SERIES}-el{$DVER}-")
+        TAG_NAMES="${REPO} ${REPO}-${TIMESTAMP}"
         tag_list=''
         for tag_prefix in "${TAG_PREFIXES[@]}"; do
-            for base in $TAG_NAMES; do
-                tag="${GITHUB_REPOSITORY/opensciencegrid\/docker-/opensciencegrid/}:$tag_prefix$base"
+            for base in ${TAG_NAMES}; do
+                tag="${IMAGE_REPO}:${tag_prefix}${base}"
                 tag_list="${tag_list}${tag_list:+,}${tag}"
             done
         done
-        echo "::set-output name=taglist::$tag_list"
+        if [[ $DVER == 7 ]] && [[ $REPO == 'testing' ]] && [[ $SERIES == '3.5' ]]; then
+            tag_list="${tag_list},$IMAGE_REPO:fresh,$IMAGE_REPO:$TIMESTAMP"
+        fi
+        echo "::set-output name=taglist::${tag_list}"
```

There are a couple broken-looking variables (`"{$SERIES}-el{$DVER}-"`), but otherwise I like the direction this is going.

Github doesn't like me making suggestions that included non-patch lines from your PR, but i am going to try it anyway

```suggestion
        IMAGE_REPO="${GITHUB_REPOSITORY/opensciencegrid\/docker-/opensciencegrid/}"
        tags=( $IMAGE_REPO:$SERIES-el$DVER-$REPO{,-$TIMESTAMP} )
        if [[ $SERIES-el$DVER-$REPO = 3.5-el7-testing ]]; then
            tags+=( $IMAGE_REPO:{fresh,$TIMESTAMP} )
        fi
        tag_list=$(IFS=,; echo "${tags[*]}")
        echo "::set-output name=taglist::${tag_list}"
```

---

Ugh, yeah, github doesn't seem to display the suggestion properly when it is across unchanged lines.

But the point is i think your `${tag_list}` can be generated with just

```
        tags=( $IMAGE_REPO:$SERIES-el$DVER-$REPO{,-$TIMESTAMP} )
        if [[ $SERIES-el$DVER-$REPO = 3.5-el7-testing ]]; then
            tags+=( $IMAGE_REPO:{fresh,$TIMESTAMP} )
        fi
        tag_list=$(IFS=,; echo "${tags[*]}")
```

---

Yeah, I was thinking last night how the outer for loop was unnecessary after I logged off last night. This looks much better!